File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    let result = '';
    const foods = food.map(el => el % 2 === 1 ? (el - 1) / 2 : el / 2);
    foods.forEach((el, idx) => el !== 0 ? result += String(idx).repeat(el) : result += '')

    return result + '0' + result.split('').reverse().join('')
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    result = '';
    foods = list(map(lambda el: (el - 1) // 2 if el % 2 == 1 else el // 2, food));
    for i in range(len(foods)):
        if foods[i] != 0:
            result += str(i) * foods[i];

    return result + '0' + result[::-1]